woocommerce/woocommerce
# Add meta table to meta table migration and enable it for WP postmeta table to order meta table.

### All Submissions:






### Changes proposed in this Pull Request:



This PR adds migrations for meta table to meta table, similar to the work we did in #32034 for meta to custom table migration. This adds a generic `MetaToMetaTableMigrator` class which implements the core of migration and then is used in `WPPostMetaToOrderMetaMigrator` class  to implement migration from wp_postmeta table to wc_order_meta table.

Closes #31768

### How to test the changes in this Pull Request:

1. Make sure that you have {prefix}_wc_orders, {prefix}_wc_order_addresses, {prefix}_wc_order_operational_data and   {prefix}_wc_orders_meta table created by following steps from https://github.com/woocommerce/woocommerce/pull/31811.
2. For a few orders, add some custom meta data.
3. Run the migration via this snippet:
```php
namespace Automattic\WooCommerce\Database\Migrations\CustomOrderTable;
$done = false;
$migrator = new WPPostToCOTMigrator();
delete_option( 'wc_cot_migration' );
$start_time = microtime( true );
while( ! $done ) {
	$batch_start = microtime( true );
	$done = $migrator->process_next_migration_batch( 500 );
	echo 'Batch completed in ' . ( microtime( true ) - $batch_start ) . ' seconds' . "\n";
}
$duration = microtime( true ) - $start_time;
echo $duration;
```
7. You can also run `delete_option( 'wc_cot_migration' );` and then migration code again to make sure that no records are duplicated.

### Other information:




### FOR PR REVIEWER ONLY:
